src/Fields/regridding_fields.jl

```diff
@@ -67,20 +67,53 @@ function we_can_regrid_in_y(a, target_grid, source_grid, b)
     return false
 end
 
+function we_can_regrid_in_x(a, target_grid, source_grid, b)
+    # Check that
+    #   1. source and target grid are in the same "class" and
+    #   2. source and target Field have same yz size
+    typeof(source_grid).name.wrapper === typeof(target_grid).name.wrapper &&
+        size(a)[[2, 3]] === size(b)[[2, 3]] && return true
+
+    return false
+end
+
+function regrid_in_z!(a, target_grid, source_grid, b)
+    arch = architecture(a)
+    source_z_faces = znodes(source_grid, Face())
+    event = launch!(arch, target_grid, :xy, _regrid_in_z!, a, b, target_grid, source_grid, source_z_faces)
+    wait(device(arch), event)
+    return a
+end
+
+function regrid_in_y!(a, target_grid, source_grid, b)
+    arch = architecture(a)
+    source_y_faces = ynodes(source_grid, Face())
+    event = launch!(arch, target_grid, :xz, _regrid_in_y!, a, b, target_grid, source_grid, source_y_faces)
+    wait(device(arch), event)
+    return a
+end
+
+function regrid_in_x!(a, target_grid, source_grid, b)
+    arch = architecture(a)
+    source_x_faces = xnodes(source_grid, Face())
+    event = launch!(arch, target_grid, :yz, _regrid_in_x!, a, b, target_grid, source_grid, source_x_faces)
+    wait(device(arch), event)
+    return a
+end
+
+regrid_in_x!(a, b) = regrid_in_x!(a, a.grid, b.grid, b)
+regrid_in_y!(a, b) = regrid_in_y!(a, a.grid, b.grid, b)
+regrid_in_z!(a, b) = regrid_in_z!(a, a.grid, b.grid, b)
 
 function regrid!(a, target_grid, source_grid, b)
```

Comment:
maybe we can change this such that it does all the regridding sequentially if we can do more than one?

Reply:
You mean, we can support regridding in more than one direction? I think you're right that we can make the intermediate grids automagically. But it's a little bit of work, I suggest we keep it for future PR